lib/opFns.js

```diff
@@ -288,6 +288,33 @@ module.exports = {
       cb(null)
     })
   },
+  EXTCODEHASH: function (address, runState, cb) {
+    if (!runState._common.gteHardfork('constantinople')) {
+      trap(ERROR.INVALID_OPCODE)
+    }
+    var stateManager = runState.stateManager
+    address = addressToBuffer(address)
+
+    if (runState._precompiled[address.toString('hex')]) {
+      return cb(null, new BN(0))
+    }
+
+    stateManager.exists(address, function (err, exists) {
+      if (err) return cb(err)
+      if (!exists) {
+        return cb(null, new BN(0))
+      }
```

Comment:
Hi @mattdean-digicatapult, short note: what occurred to mean when having a short look on how to update here regarding ``StateManager``: we should make it clear in the docs especially for the the get* methods (so ``getAccount``, ``getContractStorage``,...) what the behavior is when called on a non-existing account (will it throw or not, what is returned).

Think it would also not be overblown to have additional test cases for that. If you don't want to do yourself you can also alternatively open an issue on that.

Reply:
Thanks. 😀